microprofile/telemetry-smallrye/extension/src/main/java/org/wildfly/extension/microprofile/telemetry/MicroProfileTelemetrySubsystemAdd.java

```diff
@@ -42,7 +55,7 @@ public void execute(DeploymentProcessorTarget processorTarget) {
                         MicroProfileTelemetryExtension.SUBSYSTEM_NAME,
                         Phase.POST_MODULE,
                         Phase.POST_MODULE_MICROPROFILE_TELEMETRY,
-                        new MicroProfileTelemetryDeploymentProcessor());
+                        new MicroProfileTelemetryDeploymentProcessor(openTelemetryConfig::get));
```

Comment:
This isn't safe. AFAICT, there's no happens-before / happens-after relationship between the captor passed to L40 being invoked and this supplier being invoked. For it to be safe something would need to establish a dependency relationship between the DeploymentUnitPhaseService for Phase.POST_MODULE and the service installed at L42.

To add such a happens-before/happens-after you'd need to give your service a name and use a DUP running in a phase before POST_MODULE to add a dep on it to the next phase's DUPS. It's a bit kludgy but MicroProfileTelemetryDependencyProcessor could do it.

In practice this is quite unlikely to fail because of all the work that happens between L42 and when the DeploymentUnitPhaseService for Phase.POST_MODULE starts and calls MicroProfileTelemetryDeploymentProcessor

Reply:
I thought that was how @pferraro said to handle that, though I can't find that discussion. Do you remember, Paul?

Reply:
@jasondlee I see you added a ServiceName. Adding a couple lines to MicroProfileTelemetryDependencyProcessor.deploy will ensure the service is started before MicroProfileTelemetryDeploymentProcessor tries to read the value.

```
    @Override
    public void deploy(DeploymentPhaseContext phaseContext) {
        addDependencies(phaseContext.getDeploymentUnit());

        // Ensure the OpenTelemetryConfig is available before the next phase DeploymentUnitPhaseService starts
        phaseContext.requires(ServiceDependency.on(ServiceName.parse(WildFlyOpenTelemetryConfig.SERVICE_DESCRIPTOR.getName())));
    }
```

Reply:
Added, building, and testing...

Reply:
All clear locally. Pushed.